File: src/main/java/vini2003/xyz/bambootweaks/registry/common/BambooTweaksItems.java
package vini2003.xyz.bambootweaks.registry.common;

import vini2003.xyz.bambootweaks.BambooTweaks;
import net.minecraft.item.Item;
import net.minecraft.item.WallStandingBlockItem;
import net.minecraft.util.registry.Registry;

public class BambooTweaksItems {
	public static final Item BAMBOO_TORCH = register("bamboo_torch", new WallStandingBlockItem(BambooTweaksBlocks.BAMBOO_TORCH_BLOCK, BambooTweaksBlocks.BAMBOO_TORCH_BLOCK_WALL, new Item.Settings().group(BambooTweaksItemGroups.BAMBOO_TWEAKS)));
	public static final Item DRIED_BAMBOO_TORCH = register("dried_bamboo_torch", new WallStandingBlockItem(BambooTweaksBlocks.DRIED_BAMBOO_TORCH_BLOCK, BambooTweaksBlocks.DRIED_BAMBOO_TORCH_BLOCK_WALL, new Item.Settings().group(BambooTweaksItemGroups.BAMBOO_TWEAKS)));
	
	public static final Item DRIED_BAMBOO = register("dried_bamboo", new Item(new Item.Settings().group(BambooTweaksItemGroups.BAMBOO_TWEAKS)));
	
	public BambooTweaksItems() {
		// NO-OP
	}

	public static void initialize() {

	}

	public static <T extends Item> T register(String name, T item) {
		return Registry.register(Registry.ITEM, BambooTweaks.identifier(name), item);
	}
}

File: common/src/main/kotlin/vini2003/xyz/bambootweaks/common/registry/BambooTweaksItems.kt
package vini2003.xyz.bambootweaks.common.registry

import me.shedaniel.architectury.registry.DeferredRegister
import me.shedaniel.architectury.registry.RegistrySupplier
import net.minecraft.item.Item
import net.minecraft.item.WallStandingBlockItem
import net.minecraft.util.registry.Registry
import vini2003.xyz.bambootweaks.CommonBambooTweaks

object BambooTweaksItems {
	val ItemRegistry = DeferredRegister.create(CommonBambooTweaks.Id, Registry.ITEM_KEY)
	
	fun init() {
		ItemRegistry.register()
	}
	
	fun <T : Item> register(name: String, itemSupplier: () -> T): RegistrySupplier<T> {
		return ItemRegistry.register(CommonBambooTweaks.identifier(name), itemSupplier)
	}
	
	private fun itemSettings() = Item.Settings().group(BambooTweaksItemGroups.BambooTweaks)!!
	
	val BambooTorch = register(
		"bamboo_torch") {
		WallStandingBlockItem(
			BambooTweaksBlocks.BambooTorch.orNull!!,
			BambooTweaksBlocks.BambooWallTorch.orNull!!,
			itemSettings()
		)
	}
	
	val DriedBambooTorch = register(
		"dried_bamboo_torch") {
		WallStandingBlockItem(
			BambooTweaksBlocks.DriedBambooTorch.orNull!!,
			BambooTweaksBlocks.DriedBambooWallTorch.orNull!!,
			itemSettings()
		)
	}
	
	val DriedBamboo = register(
		"dried_bamboo") {
		Item(itemSettings())
	}
}